File: app/src/main/java/vcmsa/ci/quizapp/quizActivity.java
package vcmsa.ci.quizapp;

import android.app.Activity;

public class quizActivity extends Activity {
}
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
android:id="@+id/quiz_layout"
android:layout_width="match_parent"
android:layout_height="match_parent"
android:orientation="vertical"
android:gravity="center"
android:padding="24dp"
android:background="#FFFFFF">

    <TextView
android:id="@+id/questionTextView"
android:layout_width="wrap_content"
android:layout_height="wrap_content"
android:text="Question goes here"
android:textSize="20sp"
android:textColor="#000000"
android:gravity="center"
android:layout_marginBottom="24dp" />

    <Button
android:id="@+id/trueButton"
android:layout_width="wrap_content"
android:layout_height="wrap_content"
android:text="True"
android:backgroundTint="#4CAF50"
android:textColor="#FFFFFF"
android:layout_marginBottom="12dp" />

    <Button
android:id="@+id/falseButton"
android:layout_width="wrap_content"
android:layout_height="wrap_content"
android:text="False"
android:backgroundTint="#F44336"
android:textColor="#FFFFFF"
android:layout_marginBottom="24dp" />

    <TextView
android:id="@+id/feedbackTextView"
android:layout_width="wrap_content"
android:layout_height="wrap_content"
android:text=""
android:textSize="16sp"
android:textColor="#000000"
android:layout_marginBottom="24dp" />

    <Button
android:id="@+id/nextButton"
android:layout_width="wrap_content"
android:layout_height="wrap_content"
android:text="Next"
android:backgroundTint="#6200EE"
android:textColor="#FFFFFF"
android:enabled="false" />
</LinearLayout>

File: app/src/main/java/vcmsa/ci/quizapp/quizActivity.kt
package vcmsa.ci.quizapp

import android.annotation.SuppressLint
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import android.widget.Toast
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class QuizActivity : AppCompatActivity() {

    private val questions = arrayOf(
        "The Great Wall Of China is visible from space",
        "The capital of Australia is Sydney",
        "Lightning never strikes the same place twice",
        "Gold fish have a memory span of only three seconds",
        "The tallest mountain on Earth is Mount Everest"
    )

    private val answers = booleanArrayOf(
        false,
        false,
        false,
        false,
        true,
    )

    private var currentQuestionIndex = 0
    private var score = 0

    // View references
    private lateinit var questionTextView: TextView
    private lateinit var feedbackTextView: TextView
    private lateinit var trueButton: Button
    private lateinit var falseButton: Button

    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_quiz) // Ensure this XML is in res/layout/

        // Initialize views
        questionTextView = findViewById(R.id.questionTextView)
        feedbackTextView = findViewById(R.id.feedbackTextView)
        trueButton = findViewById(R.id.trueButton)
        falseButton = findViewById(R.id.falseButton)

        loadQuestion()

        trueButton.setOnClickListener {
            checkAnswer(true)
        }

        falseButton.setOnClickListener {
            checkAnswer(false)
        }
    }

    private fun loadQuestion() {
        if (currentQuestionIndex < questions.size) {
            questionTextView.text = questions[currentQuestionIndex]
            feedbackTextView.text = ""
        } else {
            questionTextView.text = "Quiz Completed!"
            feedbackTextView.text = "Your score: $score / ${questions.size}"
            trueButton.isEnabled = false
            falseButton.isEnabled = false
        }
    }

    private fun checkAnswer(userAnswer: Boolean) {
        val correctAnswer = answers[currentQuestionIndex]
        if (userAnswer == correctAnswer) {
            feedbackTextView.text = "Correct!"
            score++
        } else {
            feedbackTextView.text = "Incorrect!"
        }
        currentQuestionIndex++
        trueButton.postDelayed({ loadQuestion() }, 1000)
    }
}